### DISCUSSION TITLE
flutter build ipa fails during archive but fine locally

### DISCUSSION THREAD

**Discussion by xni06** [codemagic-ci-cd/codemagic-docs#2131] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Run `codemagic.yaml` workflow or the following 

```sh
flutter build ipa --release \
  --flavor development \
  --target lib/main_development.dart
```

### Expected results

Locally, this is what I get:

```sh
flutter build ipa --release \
  --flavor development \
  --target lib/main_development.dart

Archiving com.apexnetworks.ags.dev...
Automatically signing iOS for device deployment using specified development team in Xcode project: 77T825ETB5
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        13.7s
Xcode archive done.                                         46.2s
Built /Users/rhisiart/dev/apex/ags/build/ios/archive/Runner.xcarchive.

[✓] App Settings Validation
    • Version Number: 0.0.0
    • Build Number: 1
    • Display Name: AGS [DEV]
    • Deployment Target: 12.0
    • Bundle Identifier: com.apexnetworks.ags.dev

To update the settings, please refer to https://docs.flutter.dev/deployment/ios

Building App Store IPA...                                          62.1s
Built IPA to /Users/rhisiart/dev/apex/ags/build/ios/ipa.
To upload to the App Store either:
    1. Drag and drop the "build/ios/ipa/*.ipa" bundle into the Apple Transporter macOS app https://apps.apple.com/us/app/transporter/id1450874784
    2. Run "xcrun altool --upload-app --type ios -f build/ios/ipa/*.ipa --apiKey your_api_key --apiIssuer your_issuer_id".
       See "man altool" for details about how to authenticate with the App Store Connect API key.
```

### Actual results

From a Codemagic `SSH/VNC` session following a failed workflow, I see this:

```sh
builder@builders-Virtual-Machine clone % flutter build ipa --release \
  --flavor development \
  --target lib/main_development.dart

Archiving com.apexnetworks.ags.dev...
Automatically signing iOS for device deployment using specified development team in Xcode project: 77T825ETB5
Running Xcode build...                                                  
Xcode archive done.                                         26.0s
Failed to build iOS app
Error (Xcode): Target release_unpack_ios failed: Exception: Failed to codesign
/Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveInter
mediates/development/BuildProductsPath/Release-development-iphoneos/Flutter.framework/Flutter with identity
D1E4A2FE82769969C9150B935D77ED1819351CE7.


Encountered error while archiving for device.
```

... while in the actual workflow logs I see this: [flutter_build_ipa (3).log](https://github.com/codemagic-ci-cd/codemagic-docs/files/13045334/flutter_build_ipa.3.log) 

### Build id (optional)

653159aecb8601ab40f8c928

**Comment by PeetoomHeida** [comment]
Did you end up figuring this out? I am running into a similar error and am having trouble fixing it

**Reply by xni06** [reply]
I've since moved away from Codemagic and instead use GitHub Actions and Fastlane where I have more control over the environment

**Reply by helinanever** [reply]
HI @PeetoomHeida, please start a new discussion for your issue and our support team will look into it.
